opal/util/ethtool.c

```diff
@@ -0,0 +1,66 @@
+#include "opal_config.h"
+
+#include <string.h>
+#include <limits.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+#ifdef HAVE_NET_IF_H
+#include <net/if.h>
+#endif
+#ifdef HAVE_LINUX_ETHTOOL_H
+#include <linux/ethtool.h>
+#endif
+#ifdef HAVE_SYS_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+#ifdef HAVE_LINUX_SOCKIOS_H
+#include <linux/sockios.h>
+#endif
+
+#include "opal/util/ethtool.h"
+#include "opal/util/if.h"
+
+/*
+ * Set an appropriate bandwidth for the btl instance. On Linux, we obtain this
+ * via an ioctl(). Elsewhere or in the error case, we return the speed as 0.
```

Comment:
Minor nit: don't mention the BTL in the comment.  This routine can/will be used elsewhere.